tests/server.py

```diff
@@ -455,25 +464,44 @@ def getHostByName(self, name, timeout=None):
     def installNameResolver(self, resolver: IHostnameResolver) -> IHostnameResolver:
         raise NotImplementedError()
 
-    def listenUDP(self, port, protocol, interface="", maxPacketSize=8196):
+    def listenUDP(
+        self,
+        port: int,
+        protocol: DatagramProtocol,
+        interface: str = "",
+        maxPacketSize: int = 8196,
+    ) -> udp.Port:
         p = udp.Port(port, protocol, interface, maxPacketSize, self)
         p.startListening()
         self._udp.append(p)
         return p
 
-    def callFromThread(self, callback, *args, **kwargs):
+    def callFromThread(
+        self, callable: Callable[..., Any], *args: object, **kwargs: object
+    ) -> None:
```

Comment:
Can I persuade you to make `callable: Callable[P, R]` for some `ParamSpec` P, and then `*args: P.args` and `**kwargs: P.kwargs`? (Maybe the same for `callInThread` below?

Reply:
This is what I originally had, but it doesn't match the interface if you do that.

Reply:
Ah, that's a shame. An opportunity to improve twisted's annotations, I guess!